### DISCUSSION TITLE
Platipus loss function potentially doesn't match paper

### DISCUSSION THREAD

**Discussion by ju-leon** [cnguyen10/few_shot_meta_learning#15] [Q&A]
Hi,

thank you for the great implementation of meta learning algorithms.

We are trying to evaluate the PLATIPUS algorithm and we noted that the paper requires a Negative Log Likelihood loss (Page 3, 3 Preliminaries), yet your adaptation step by default is using an MSE loss.

We were wondering if this is an adaptation of the original paper on your part or if we’re missing a crucial step where the NLL is calculated from the MSE. Later the loss we’re suspecting to be an MSE loss is logged to Tensorboard as a NLL again(`Platipus.py, 216`). This seems especially important since in `Platipus.py, 184` the gradient is calculated on this loss function. To our understanding, this gradient might differ significantly from the gradient intended in the paper since it is calculated on a different loss function. 

Is this just a trick in the implementation to use the MSE instead of the NLL or are we missing something in the implementation?

Thanks a lot in advance,
Leon

**Comment by cnguyen10** [comment]
I am not sure if I understand your question. Negative log-likelihood is an "abstract" loss function, e.g. it will be MSE in regression (line 198 in ```main.py```) corresponding to a Gaussian likelihood or Cross-entropy loss in classification (line 185 in ```main.py```) corresponding to a Categorical likelihood. In the ```Platipus.py```, the loss function is defined consistently with the configuration specified in ```main.py``` (see line 19 of the constructor in ```Platipus.py``` where ```config``` is copied to ```self.config```). Thus, when you run regression, the NLL display on the Tensorboard is MSE.

**Comment by ju-leon** [comment]
Thank you for your explanation!
After going back to the paper we could sort out our confusion and connect MSE and NLL.
So the issue was on our side and not in your implementation.
